### ISSUE TITLE
Is it possible run omnisharp-vscode without .NET CLI tools?

### ISSUE THREAD

**Issue by adamstyl** [OmniSharp/omnisharp-vscode#1887]
## Environment data
`dotnet --info` output:
'dotnet' is not recognized as an internal or external command,
operable program or batch file.

VS Code version: 1.18.1
C# Extension version: 1.13.1

First of all apologies for hijacking the issue tracker to post a question/request. My problem is pretty much described int [this](https://stackoverflow.com/questions/46936605/is-it-possible-to-use-visual-studio-code-with-c-sharp-plugin-in-windows-without) stackoverflow question.

The TL;DR is that I don't have admin access to install .NET CLI tools and I would like to know if it is possible to run omnisharp-vscode without them. Jetbrains Rider is able to do it (tested it on the same machine) and I _think_ it leverages Roslyn to do that, but I'm no sure. So there is _some_ way, but I don't know if that is possible with omnisharp-vscode.

**Comment by gregg-miskelly** [comment]
If you just want editing - you can ignore this message. There is message is just trying to tell you that .NET Core debugging/building will not work. There is a preference to disable it.

If you want .NET Core debugging + building: all the extension does is invoke 'dotnet' on the command line. I haven't tried this, but I don't know of any reason why you couldn't download the .NET CLI zips, extract them, and update your `PATH` environment variable to include the directory where you extracted things to.

**Comment by adamstyl** [comment]
Hi @gregg-miskelly and thanks for the reply. No I don't want "just editing" I also want navigation: Ctrl + click, Ctrl + T, refactoring, the whole nine yards. But this seems impossible with the errors I get. And I don't think that I can change PATH on a computer I don't have administrative rights although I didn't try it myself.

Please have a look when time permits to my question in stackoverflow, I have included many more details there.

**Comment by gregg-miskelly** [comment]
@adamstyl every user account, and for that matter every process contains its own PATH. So yes, you can definitely change the path without admin rights. For a quick experiment, just open a command prompt and run `set PATH=%PATH%;c:\your\new\directory\goes\here`. Then start VS Code from that command prompt.

**Comment by adamstyl** [comment]
@gregg-miskelly  Thanks, I will try that. Can you please share a link for the CLI _zip_? I clicked on "Get .NET CLI tools" and it got me to https://www.microsoft.com/net/learn/get-started/windows#windowscmd where there is only an exe, no zip :(

**Comment by gregg-miskelly** [comment]
https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.3.md

**Comment by adamstyl** [comment]
Good news and bad news. I got the CLI tools working. The editor is still a fancy notepad though; nothing close to an IDE. No Ctrl + click to navigate, nothing. And the error remains. I will post it here although it is already in the stackoverflow question that I've linked in my first post:

> 
> [fail]: OmniSharp.MSBuild.ProjectFile.ProjectFileInfo
>         The reference assemblies for framework ".NETFramework,Version=v4.0" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend.

**Comment by DustinCampbell** [comment]
@adamstyl : I'm not sure why you closed this issue since it's still not working for you. :smile:

A couple of questions:

* What operating system are you running? Windows?
* Could you provide your OmniSharp Log? You can retrieve this by selecting View->Output from the VS Code menu and then selecting "OmniSharp Log" from the drop-down in the top-right corner.

**Comment by DustinCampbell** [comment]
Sorry, I see that you filed a separate issue.

**Comment by adamstyl** [comment]
Hi @DustinCampbell  indeed I should have been a bit more explicit here and mention that part II of the problem is in another issue 😃
